lib/private/User/Database.php

```diff
@@ -368,46 +369,62 @@ public function checkPassword(string $loginName, string $password) {
 	/**
 	 * Load an user in the cache
 	 *
-	 * @param string $uid the username
+	 * @param string $loginName the username or email
 	 * @return boolean true if user was found, false otherwise
 	 */
-	private function loadUser($uid) {
-		$this->fixDI();
+	private function loadUser(string $loginName, bool $tryEmail = true): bool {
+		if (isset($this->cache[$loginName])) {
+			return true;
+		}
 
-		$uid = (string)$uid;
-		if (!isset($this->cache[$uid])) {
-			//guests $uid could be NULL or ''
-			if ($uid === '') {
-				$this->cache[$uid] = false;
-				return true;
-			}
+		//guests $uid could be NULL or ''
+		if ($loginName === '') {
+			$this->cache[$loginName] = false;
+			return true;
```

Comment:
according to phpdoc return true here means user was found, but here it was not?

Reply:
I think this was a hack for guest users but seems not needed anymore.
Ref: https://github.com/nextcloud/server/pull/3007

```suggestion
			return false;
```